### ISSUE TITLE
Using ControlFlow::WaitUntil causes unpredictable rendering, regardless of placement

### ISSUE THREAD

**Issue by campbellcole** [glium/glium#1928]
I have been writing a test program and it seems no matter where I place the `*control_flow = glutin::event_loop::ControlFlow::WaitUntil()` assignment, the FPS fluctuates when the mouse moves. I have tried placing it before and after the `match event (glutin::event::Event)`, before and after the `match event (glutin::event::WindowEvent)` and inside of the `glutin::event::Event::MainEventsCleared` match case to no avail.

Without the control flow wait, I average over 600 FPS, and with it (with no mouse movement), I average around 30-40 FPS. While the mouse is moving the FPS rises to just over 100 FPS. I am calling the `WaitUntil` function with the arguments supplied by the tutorial (`16_666_667` nanos), and it seems that somehow the draw command gets pushed without ever calling the control flow assignment (when I've placed it inside the `MainEventsCleared` case). Any help would be appreciated.

I noticed the most recent commits for some of the tutorials claims to fix this problem, but using an identical method in my code does not seem to work.

**Comment by campbellcole** [comment]
After doing some research, I found that the tutorials use what a glutin developer considered a [bad method of controlling the framerate](https://github.com/rust-windowing/glutin/issues/1336#issuecomment-721368072). The way I solved this issue was by using `let cb = glutin::ContextBuilder::new().with_vsync(true);`.

I guess this is now an issue with the tutorials, but is no longer relevant so I will close it. Perhaps consider changing the tutorial, as a developer explicitly said the method used is not a good way of doing this.
